test/e2e/rc.go

```diff
@@ -235,3 +252,91 @@ func rcConditionCheck(f *framework.Framework) {
 	}
 	Expect(err).NotTo(HaveOccurred())
 }
+
+func testRCAdoptMatchingOrphans(f *framework.Framework) {
+	name := "pod-adoption"
+	By(fmt.Sprintf("Given a Pod with a 'name' label %s is created", name))
+	p := f.PodClient().CreateSync(&v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+			Labels: map[string]string{
+				"name": name,
+			},
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{
+					Name:  name,
+					Image: nginxImageName,
+				},
+			},
+		},
+	})
+
+	By("When a replication controller with a matching selector is created")
+	replicas := int32(2)
+	rcSt := newRCWithSelector(name, replicas, map[string]string{"name": name}, name, nginxImageName, name)
+	rc, err := f.ClientSet.Core().ReplicationControllers(f.Namespace.Name).Create(rcSt)
+	Expect(err).NotTo(HaveOccurred())
+	// Cleanup the ReplicationController when we are done.
+	defer func() {
```

Comment:
No need for this function since the namespace deletion will clean the resources up. I am removing it from the rest of the tests in https://github.com/kubernetes/kubernetes/pull/42996